src/Workspaces/Core/Portable/Workspace/Solution/SolutionCompilationState.cs

```diff
@@ -53,11 +53,14 @@ internal sealed partial class SolutionCompilationState
     private ConditionalWeakTable<ISymbol, ProjectId?>? _unrootedSymbolToProjectId;
     private static readonly Func<ConditionalWeakTable<ISymbol, ProjectId?>> s_createTable = () => new ConditionalWeakTable<ISymbol, ProjectId?>();
 
+    private readonly AsyncLazy<SolutionCompilationState> _cachedFrozenSnapshot;
```

Comment:
Isn't this just undoing https://github.com/dotnet/roslyn/pull/72193? Becuase I liked that one. I didn't undestand why we had AsyncLazies at both levels.

Reply:
it's not *quite* undoing it.  the cachgin, and what we hand out, is still cached Solution instances (which is good for downstream consumers).  However, under the covers there are a couple of paths that get to asking for the frozen compilation state.  And this ensures they can happen and still get the same result.  this is not really that necessary though, and i will considering removing it.  i just very much liked the idea that sharing woudl happen if possible.